hooks/useClickOutside.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useRef } from 'react';
+
+type useClickOutsideType = (
+  // eslint-disable-next-line no-unused-vars
+  event: MouseEvent | TouchEvent
+) => void;
+
+export const useClickOutside = <T extends HTMLElement>(
+  handler: useClickOutsideType
+) => {
+  const ref = useRef<T>(null);
+  useEffect(() => {
+    const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'click';
+    const listener = (event: MouseEvent | TouchEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) {
```

Comment:
I still rather not have negated logic, this is over complex.

Reply:
I turned the problem in all directions and by spanking the condition with "?" it does not return when you click on the ref

Reply:
I'm actually okay with the way @AugustinMauroy has implemented it.

The reason we can't combine these two conditions is because one is checking for negation, while the other is not checking for that, and both conditions are valid. 

I believe reading it the way it's written right now is much more simple and easy to understand, but combining them with an optional chaining operator makes it more complex to understand what's going on. 

I pointed out the mistake in the previous one here, which is why the tests were failing:
https://github.com/nodejs/nodejs.org/pull/5359#discussion_r1186873691

Reply:
Maybe we could change it to this to make it more clear:
```typescript
if (ref.current && !ref.current.contains(event.target as Node)) {
  handler(event);
}
```

Reply:
So I thinks we are good now